src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java

```diff
@@ -1,97 +1,250 @@
-
-@ConfigureContext(session = UserNameFixture.mo14)
-public class CuCreditMemoServiceImplTest extends KualiTestBase {
-
-	private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CuCreditMemoServiceImplTest.class);
-
-	private CuCreditMemoServiceImpl creditMemoServiceImpl;
-	private DocumentService documentService;
-
-	@Override
-	protected void setUp() throws Exception {
-		super.setUp();
-
-		creditMemoServiceImpl = (CuCreditMemoServiceImpl) TestUtils.getUnproxiedService("creditMemoService");
-		documentService = SpringContext.getBean(DocumentService.class);
-
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.core.impl.datetime.DateTimeServiceImpl;
+import org.kuali.rice.kew.api.exception.WorkflowException;
+import org.kuali.rice.kim.api.identity.Person;
+import org.powermock.api.easymock.PowerMock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(CuCreditMemoServiceImplTest.TestCuCreditMemoServiceImpl.class)
+public class CuCreditMemoServiceImplTest {
+
+    private AccountsPayableService accountsPayableService;
+    private CuCreditMemoServiceImpl creditMemoServiceImpl;
+    private DataDictionaryService dataDictionaryService;
+    private DateTimeService dateTimeService;
+    private DocumentService documentService;
+    private NoteService noteService;
+    private PurapService purapService;
+    private VendorService vendorService;
+    private CuVendorCreditMemoDocument creditMemoDocument;
+    @Mock
+    private static Person mo14Person;
+    @Mock
+    private static UserSession mo14Session;
+
+    @Before
+	public void setUp() throws Exception {
+        accountsPayableService = EasyMock.createNiceMock(AccountsPayableService.class);
+        EasyMock.expect(accountsPayableService.getExpiredOrClosedAccountList(creditMemoDocument)).andReturn(new HashMap<>());
+        EasyMock.replay(accountsPayableService);
+
+        dataDictionaryService = EasyMock.createNiceMock(DataDictionaryService.class);
+        EasyMock.expect(dataDictionaryService.getAttributeMaxLength(DocumentHeader.class, KRADPropertyConstants.DOCUMENT_DESCRIPTION)).andReturn(200);
+        EasyMock.replay(dataDictionaryService);
+
+        dateTimeService = new DateTimeServiceImpl();
+        documentService = new MockDocumentServiceImpl();
+        noteService = EasyMock.createMock(NoteService.class);
+        purapService = EasyMock.createMock(PurapService.class);
+        vendorService = new MockVendorServiceImpl();
+
+        creditMemoServiceImpl = PowerMock.createPartialMock(CuCreditMemoServiceImplTest.TestCuCreditMemoServiceImpl.class, "reIndexDocument", "getCreditMemoDocumentById");
```

Comment:
The mocked creditMemoServiceImpl is only being replayed by some of the unit test methods.